test/test_progress.py

```diff
@@ -0,0 +1,63 @@
+######################################################################
+#
+# File: test/test_base.py
+#
+# Copyright 2017, Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from b2.progress import StreamWithHash
+from b2.utils import hex_sha1_of_bytes
+import six
+from .test_base import TestBase
+
+
+class TestHashingStream(TestBase):
+    def setUp(self):
+        self.data = b'01234567'
+        self.stream = StreamWithHash(six.BytesIO(self.data))
+        self.hash = hex_sha1_of_bytes(self.data)
+        self.expected = self.data + self.hash.encode()
+
+    def test_no_argument(self):
+        output = self.stream.read()
+        self.assertEqual(self.expected, output)
+
+    def test_no_argument_less(self):
+        output = self.stream.read(len(self.data) - 1)
+        output += self.stream.read()
+        self.assertEqual(self.expected, output)
+
+    def test_no_argument_equal(self):
+        output = self.stream.read(len(self.data))
+        output += self.stream.read()
+        self.assertEqual(self.expected, output)
+
+    def test_no_argument_more(self):
+        output = self.stream.read(len(self.data) + 1)
+        output += self.stream.read()
+        self.assertEqual(self.expected, output)
+
+    def test_one_by_one(self):
+        output = b''
+        data = self.stream.read(1)
```

Comment:
How do I write this for Python 2 and 3? With your version I get assertion failures that 48 is not equal to b'0' on Python 3. So I used `six.int2byte(expected_byte)` and now Python 2 complains that `expected_byte` is not an integer.